### ISSUE TITLE
Stack-trace didn't de-obfuscated

### ISSUE THREAD

**Issue by MohammadRezaei92** [DataDog/dd-sdk-android-gradle-plugin#32]
Thanks for taking the time for reporting an issue!

**Describe what happened**
I added gradle plugin and did the configuration. Gradle task for upload mapping file run successfully but still my stack-traces are obfuscated.

**Steps to reproduce the issue:**

**Describe what you expected:**
Seeing stack-trace de-obfuscated.

**Additional context**
 - Android Build Tools version:
 - Gradle version:4.1.0
 - OS version:
 - Plugin version:1.0.1
 - Proguard configuration: 

```
# Add project specific ProGuard rules here.
# You can control the set of applied configuration files using the
# proguardFiles setting in build.gradle.
#
# For more details, see
#   http://developer.android.com/guide/developing/tools/proguard.html

# If your project uses WebView with JS, uncomment the following
# and specify the fully qualified class name to the JavaScript interface
# class:
-keepclassmembers class fqcn.of.javascript.interface.for.webview {
   public *;
}

# Uncomment this to preserve the line number information for
# debugging stack traces.
#-keepattributes SourceFile,LineNumberTable

# If you keep the line number information, uncomment this to
# hide the original source file name.
#-renamesourcefileattribute SourceFile


# Fabric
-keepattributes *Annotation*
-keepattributes SourceFile,LineNumberTable
-keep public class * extends java.lang.Exception
#-printmapping mapping.txt

-keep class com.crashlytics.** { *; }
-dontwarn com.crashlytics.**

#Disable logging
-assumenosideeffects class android.util.Log {
    public static *** d(...);
    public static *** v(...);
    public static *** i(...);
}

### https://gist.github.com/jemshit/767ab25a9670eb0083bafa65f8d786bb

### Glide, Glide Okttp Module, Glide Transformations
-keep public class * implements com.bumptech.glide.module.GlideModule
-keep public class * extends com.bumptech.glide.module.AppGlideModule
-keep public enum com.bumptech.glide.load.resource.bitmap.ImageHeaderParser$** {
  **[] $VALUES;
  public *;
}
# -keepresourcexmlelements manifest/application/meta-data@value=GlideModule 3 For dexguard

### Viewpager indicator
-dontwarn com.viewpagerindicator.**

### Support v7, Design
# http://stackoverflow.com/questions/29679177/cardview-shadow-not-appearing-in-lollipop-after-obfuscate-with-proguard/29698051
-keep class android.support.v7.widget.RoundRectDrawable { *; }

-keep public class android.support.v7.widget.** { *; }
-keep public class android.support.v7.internal.widget.** { *; }
-keep public class android.support.v7.internal.view.menu.** { *; }

-keep public class * extends android.support.v4.view.ActionProvider {
    public <init>(android.content.Context);
}

-dontwarn android.support.**
-dontwarn android.support.design.**
-keep class android.support.design.** { *; }
-keep interface android.support.design.** { *; }
-keep public class android.support.design.R$* { *; }

# https://github.com/Gericop/Android-Support-Preference-V7-Fix/blob/master/preference-v7/proguard-rules.pro
-keepclassmembers class android.support.v7.preference.PreferenceGroupAdapter {
    private ** mPreferenceLayouts;
}
-keepclassmembers class android.support.v7.preference.PreferenceGroupAdapter$PreferenceLayout {
    private int resId;
    private int widgetResId;
}

# https://github.com/dandar3/android-support-animated-vector-drawable/blob/master/proguard-project.txt
#-keepclassmembers class android.support.graphics.drawable.VectorDrawableCompat$* {
#   void set*(***);
#   *** get*();
#}

# Gson uses generic type information stored in a class file when working with fields. Proguard
# removes such information by default, so configure it to keep all of it.
-keepattributes Signature

# For using GSON @Expose annotation
-keepattributes *Annotation*

# Gson specific classes
-keep class sun.misc.Unsafe { *; }
#-keep class com.google.gson.stream.** { *; }

# Application classes that will be serialized/deserialized over Gson
# -keep class com.google.gson.examples.android.model.** { *; }

# Prevent proguard from stripping interface information from TypeAdapterFactory,
# JsonSerializer, JsonDeserializer instances (so they can be used in @JsonAdapter)
-keep class * implements com.google.gson.TypeAdapterFactory
-keep class * implements com.google.gson.JsonSerializer
-keep class * implements com.google.gson.JsonDeserializer

### Retrofit 2
# Platform calls Class.forName on types which do not exist on Android to determine platform.
-dontnote retrofit2.Platform
# Platform used when running on RoboVM on iOS. Will not be used at runtime.
-dontnote retrofit2.Platform$IOS$MainThreadExecutor
# Platform used when running on Java 8 VMs. Will not be used at runtime.
-dontwarn retrofit2.Platform$Java8
# Retain generic type information for use by reflection by converters and adapters.
-keepattributes Signature
# Retain declared checked exceptions for use by a Proxy instance.
-keepattributes Exceptions

-dontwarn retrofit2.adapter.rxjava.CompletableHelper$** # https://github.com/square/retrofit/issues/2034
#To use Single instead of Observable in Retrofit interface
-keepnames class rx.Single
#Retrofit does reflection on generic parameters. InnerClasses is required to use Signature and
# EnclosingMethod is required to use InnerClasses.
-keepattributes Signature, InnerClasses, EnclosingMethod
# Retain service method parameters when optimizing.
-keepclassmembers,allowshrinking,allowobfuscation interface * {
    @retrofit2.http.* <methods>;
}
# Ignore annotation used for build tooling.
-dontwarn org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
# Ignore JSR 305 annotations for embedding nullability information.
-dontwarn javax.annotation.**
# Guarded by a NoClassDefFoundError try/catch and only used when on the classpath.
-dontwarn kotlin.Unit
# Top-level functions that can only be used by Kotlin.
-dontwarn retrofit2.-KotlinExtensions

### OkHttp3
-dontwarn okhttp3.**
-dontwarn okio.**
-dontwarn javax.annotation.**
# A resource is loaded with a relative path so the package of this class must be preserved.
-keepnames class okhttp3.internal.publicsuffix.PublicSuffixDatabase


### Kotlin
#https://stackoverflow.com/questions/33547643/how-to-use-kotlin-with-proguard
#https://medium.com/@AthorNZ/kotlin-metadata-jackson-and-proguard-f64f51e5ed32
-keepclassmembers class **$WhenMappings {
    <fields>;
}
-keep class kotlin.Metadata { *; }
-keepclassmembers class kotlin.Metadata {
    public <methods>;
}

### Other
-dontwarn com.google.errorprone.annotations.*

### Android Architecture Components
# Ref: https://issuetracker.google.com/issues/62113696
# LifecycleObserver's empty constructor is considered to be unused by proguard
#-keepclassmembers class * implements android.arch.lifecycle.LifecycleObserver {
#    <init>(...);
#}
-keep class * implements android.arch.lifecycle.LifecycleObserver {
    <init>(...);
}

# ViewModel's empty constructor is considered to be unused by proguard
-keepclassmembers class * extends android.arch.lifecycle.ViewModel {
    <init>(...);
}

# keep Lifecycle State and Event enums values
-keepclassmembers class android.arch.lifecycle.Lifecycle$State { *; }
-keepclassmembers class android.arch.lifecycle.Lifecycle$Event { *; }
# keep methods annotated with @OnLifecycleEvent even if they seem to be unused
# (Mostly for LiveData.LifecycleBoundObserver.onStateChange(), but who knows)
-keepclassmembers class * {
    @android.arch.lifecycle.OnLifecycleEvent *;
}

## Databinding or library depends on databinding
-dontwarn android.databinding.**
-keep class android.databinding.** { *; }

### Kotlin Coroutine
# https://github.com/Kotlin/kotlinx.coroutines/blob/master/README.md
# ServiceLoader support
-keepnames class kotlinx.coroutines.internal.MainDispatcherFactory {}
-keepnames class kotlinx.coroutines.CoroutineExceptionHandler {}
-keepnames class kotlinx.coroutines.android.AndroidExceptionPreHandler {}
-keepnames class kotlinx.coroutines.android.AndroidDispatcherFactory {}
# Most of volatile fields are updated with AFU and should not be mangled
-keepclassmembernames class kotlinx.** {
    volatile <fields>;
}
# https://github.com/Kotlin/kotlinx.atomicfu/issues/57
-dontwarn kotlinx.atomicfu.**

### Kotlin
#https://stackoverflow.com/questions/33547643/how-to-use-kotlin-with-proguard
#https://medium.com/@AthorNZ/kotlin-metadata-jackson-and-proguard-f64f51e5ed32
-keepclassmembers class **$WhenMappings {
    <fields>;
}
-keep class kotlin.Metadata { *; }
-keepclassmembers class kotlin.Metadata {
    public <methods>;
}

### Google Play Billing
-keep class com.android.vending.billing.**

### Adjust SDK, android.installreferrer
-keep public class com.adjust.sdk.** { *; }
-keep class com.google.android.gms.common.ConnectionResult {
    int SUCCESS;
}
-keep class com.google.android.gms.ads.identifier.AdvertisingIdClient {
    com.google.android.gms.ads.identifier.AdvertisingIdClient$Info getAdvertisingIdInfo(android.content.Context);
}
-keep class com.google.android.gms.ads.identifier.AdvertisingIdClient$Info {
    java.lang.String getId();
    boolean isLimitAdTrackingEnabled();
}
-keep public class com.android.installreferrer.** { *; }

### LeakCanary
-keep class org.eclipse.mat.** { *; }
-keep class com.squareup.leakcanary.** { *; }

### Nothing for Butterknife 8, Realm, RxJava2, RxBinding, RxRelay, Dagger2, OneSignal, Google Play Services, Firebase, Facebook SDK, Room DB,

# Smartlook
-keep class com.smartlook.sdk.** { *; }

# Stadroid
-keep class io.stacrypt.stadroid.data.** { *; }
-keep class io.stacrypt.stadroid.ui.BankIIN { *; }
#-keep class io.stacrypt.stadroid.data.ModelKt.** { *; }

-keepclassmembers enum * { *; }

-keep class net.gotev.uploadservice.** { *; }

# FIXME: These lines should be remved as soon as this bug fixed on datadog's side:
# https://github.com/DataDog/dd-sdk-android/issues/362
-keepnames class * extends android.app.Activity
-keepnames class * extends androidx.fragment.app.Fragment
```

 - Other Gradle Plugins:

```
apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'
apply plugin: 'kotlin-kapt'
apply plugin: "androidx.navigation.safeargs"
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'
apply plugin: 'com.google.firebase.firebase-perf'
apply from: '../ktlint.gradle'
apply from: '../bumper.gradle'
```

**Comment by 0xnm** [comment]
Hello @MohammadRezaei92! Thanks for opening the issue. Do you see any errors/warnings related to this plugin when you run mapping upload task?

**Comment by MohammadRezaei92** [comment]
> Hello @MohammadRezaei92! Thanks for opening the issue. Do you see any errors/warnings related to this plugin when you run mapping upload task?

As I see there is no warning or error.
![image](https://user-images.githubusercontent.com/14319444/117939711-94fd8c00-b31d-11eb-8b09-c247984a26b5.png)

**Comment by 0xnm** [comment]
@MohammadRezaei92 Can you please run this task with `info` log level enabled? This should give more information. You can check how to do it here https://docs.gradle.org/current/userguide/logging.html

**Comment by MohammadRezaei92** [comment]
> @MohammadRezaei92 Can you please run this task with `info` log level enabled? This should give more information. You can check how to do it here https://docs.gradle.org/current/userguide/logging.html

![image](https://user-images.githubusercontent.com/14319444/118231065-d28c2180-b4a3-11eb-97e9-6438b63aa893.png)

**Comment by 0xnm** [comment]
@MohammadRezaei92 Thanks for the info! Can you please confirm that you see non-obfuscated stacktrace in RUM (because we don't do de-obfuscation in Logs yet)? Also can you please check that service name which is used during `Datadog.initialize()` call is the same as in the plugin configuration?

**Comment by MohammadRezaei92** [comment]
> @MohammadRezaei92 Thanks for the info! Can you please confirm that you see non-obfuscated stacktrace in RUM (because we don't do de-obfuscation in Logs yet)? Also can you please check that service name which is used during `Datadog.initialize()` call is the same as in the plugin configuration?

I see stack-trace in RUM like this
![image](https://user-images.githubusercontent.com/14319444/118250977-4508fb80-b4bc-11eb-900a-8fd7e0d652c0.png)

**Comment by 0xnm** [comment]
@MohammadRezaei92 If you are using custom service name please check it matches the values used in `Datadog.initialize`, also please check that stacktrace entry is generated at the app version which had mapping file uploaded (we don't apply mapping file for the particular version to another versions). If everything looks good, please open support ticket (https://docs.datadoghq.com/help/) with the details of your App ID/Org ID, so that we are able to continue looking on the issue more precisely.

**Comment by MohammadRezaei92** [comment]
> @MohammadRezaei92 If you are using custom service name please check it matches the values used in `Datadog.initialize`, also please check that stacktrace entry is generated at the app version which had mapping file uploaded (we don't apply mapping file for the particular version to another versions). If everything looks good, please open support ticket (https://docs.datadoghq.com/help/) with the details of your App ID/Org ID, so that we are able to continue looking on the issue more precisely.

We don't use a custom service name and the version number is OK.

**Comment by 0xnm** [comment]
Hello @MohammadRezaei92. The issue is solved now and to summarize, the reason was:

* In multi-variant app, `variant` should be specified when Datadog SDK is initialized. This API is available with `dd-sdk-android >= 1.8.0`
* De-obfuscation is not yet supported in the Logs, only in the RUM. More info can be found [here](https://docs.datadoghq.com/real_user_monitoring/error_tracking/explorer).
